### PR TITLE
Document inclusion criteria for new functionality in core developer guide

### PR DESCRIPTION
**Pull request by lagru**
## Description

Our guidelines and expectations on whether to include new functionality are currently not well documented. This tries to address this situation based on discussion and feedback in [1].

[1] https://discuss.scientific-python.org/t/405

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- Descriptive commit messages (see below)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
  `doc/release/release_dev.rst`.
- There is a bot to help automate backporting a PR to an older branch. For
  example, to backport to v0.19.x after merging, add the following in a PR
  comment: `@meeseeksdev backport to v0.19.x`
- To run benchmarks on a PR, add the `run-benchmark` label. To rerun, the label
  can be removed and then added again. The benchmark output can be checked in
  the "Actions" tab.

### CODE REVIEW

File: doc/source/core_developer.md [hunk 1]
```diff
@@ -80,6 +80,17 @@ When reviewing, focus on the following:
    tests not only execute the code, but explores corner cases.  It is tempting
    not to review tests, but please do so.
 
+5. **Licensing:** New contributions _must_ be compatible with  `scikit-image`'s
```
Comment (mkcor) [inline doc/source/core_developer.md:83]:
```suggestion
5. **Licensing:** New contributions _must_ be compatible with `scikit-image`'s 
```

File: doc/source/core_developer.md [hunk 2]
```diff
@@ -80,6 +80,17 @@ When reviewing, focus on the following:
    tests not only execute the code, but explores corner cases.  It is tempting
    not to review tests, but please do so.
 
+5. **Licensing:** New contributions _must_ be compatible with  `scikit-image`'s 
+   license. Which licenses are compatible is a hard question to answer. If in
```
Comment (mkcor) [inline doc/source/core_developer.md:84]: Link to the `LICENSE.txt` file?

Comment (lagru) [reply doc/source/core_developer.md:84]: Thanks, I'm on it but conda is taking its sweet time to install the dependencies to test it locally. :sweat_smile:

**Review by jni — APPROVED** [review]
🙏

File: doc/source/core_developer.md [hunk 3]
```diff
@@ -80,6 +80,17 @@ When reviewing, focus on the following:
...
    not to review tests, but please do so.
 
+5. **Licensing:** New contributions _must_ be compatible with 
+   {doc}`scikit-image's license <license>`. Which licenses are compatible is a
+   hard question to answer. If in doubt, ask the team for help. If the 
+   contributor is not the copyright holder of the submitted code, ask for
+   approval and include them in our license.
+
+6. **Established methods:** In general, we are looking to include well cited 
+   algorithms and methods that are of general use to the imaging community.
```
Comment (alexdesiqueira) [inline doc/source/core_developer.md:90]: How about _(...) to include widely used algorithms and methods (...)_

File: doc/source/core_developer.md [hunk 4]
```diff
@@ -80,6 +80,17 @@ When reviewing, focus on the following:
    tests not only execute the code, but explores corner cases.  It is tempting
    not to review tests, but please do so.
 
+5. **Licensing:** New contributions _must_ be compatible with 
+   {doc}`scikit-image's license <license>`. Which licenses are compatible is a
+   hard question to answer. If in doubt, ask the team for help. If the
```
Comment (alexdesiqueira) [inline doc/source/core_developer.md:85]: How about _(...) contributions should be available in the same license or be compatible with (...) question to answer. Examples of BSD-compatible licenses are MIT (link) and Apache (link). If in doubt (...)_

**Comment by alexdesiqueira** [comment]
Thanks @lagru! CI error is the one being dealt with in #6490: 

```
=================================== FAILURES ===================================
_________________ [doctest] skimage.future.graph.rag.show_rag __________________
200     Examples
201     --------
202     >>> from skimage import data, segmentation
203     >>> from skimage.future import graph
204     >>> import matplotlib.pyplot as plt
205     >>>
206     >>> img = data.coffee()
207     >>> labels = segmentation.slic(img)
208     >>> g =  graph.rag_mean_color(img, labels)
209     >>> lc = graph.show_rag(labels, g, img)
UNEXPECTED EXCEPTION: ImportError('"<function show_rag at 0x7f241db1fc10>" in "skimage.future.graph.rag" requires "matplotlib >=3.5"')
```

File: doc/source/core_developer.md (hunk 4 above)
Comment (mkcor) [reply doc/source/core_developer.md:85]: +1 to @alexdesiqueira's suggestion; maybe "in the same license" --> "under the same license"

File: doc/source/core_developer.md [hunk 5]
```diff
@@ -80,8 +80,23 @@ When reviewing, focus on the following:
    tests not only execute the code, but explores corner cases.  It is tempting
    not to review tests, but please do so.
 
+5. **Licensing:** New contributions should be available under the same license
+   or be compatible with {doc}`scikit-image's license <license>`. Which ones are
+   compatible is a hard question to answer. Examples of BSD-compatible licenses
```
Comment (alexdesiqueira) [inline doc/source/core_developer.md:85]:
```suggestion
   or be compatible with {doc}`scikit-image's license <license>`. Examples of BSD-
   compatible licenses
```

File: doc/source/core_developer.md [hunk 6]
```diff
@@ -80,8 +80,23 @@ When reviewing, focus on the following:
...
+   are the [MIT License][mit_license] and [Apache License 2.0][apache_2-2]. If
+   in doubt, ask the team for help. If the contributor is not the copyright
```
Comment (alexdesiqueira) [inline doc/source/core_developer.md:87]:
```suggestion
   are the [MIT License][mit_license] and [Apache License 2.0][apache_2-2]. When
   in doubt, ask the team for help. If you, as the contributor, is not the copyright
```

File: doc/source/core_developer.md [hunk 7]
```diff
@@ -80,8 +80,23 @@ When reviewing, focus on the following:
...
+   holder of the submitted code, ask for approval and include them in our
+   license.
```
Comment (alexdesiqueira) [inline doc/source/core_developer.md:89]:
```suggestion
   holder of the submitted code, please ask for approval from the original authors and
   include their names in `LICENSE.txt`. You can use the other entries in that file as
   templates.
```

File: doc/source/core_developer.md [hunk 8]
```diff
@@ -80,8 +80,23 @@ When reviewing, focus on the following:
...
+
+6. **Established methods:** In general, we are looking to include algorithms and
+   methods that are established and widely used in literature and by the imaging 
+   community. While this is not a hard requirement, new contributions should
+   further {doc}`our mission <values>`.
```
Comment (alexdesiqueira) [inline doc/source/core_developer.md:94]:
```suggestion
   be consistent with {doc}`our mission <values>`.
```

**Review by alexdesiqueira — APPROVED** [review]
Thank you for the updates, @lagru!
Just a few more suggestions; please feel free to use them or not.
Thanks again!

**Review by mkcor — APPROVED** [review]
@lagru thank you for documenting what came up in recent discussions! I have taken the liberty of committing @alexdesiqueira's suggestions, pulling your branch, and adding a few edits. I'm not merging right now so you can proofread.

**Comment by lagru** [comment]
Thanks everyone! Merging.
